utils/gatewayEvm.ts

```diff
@@ -0,0 +1,216 @@
+import { AbiCoder, ethers } from "ethers";
+
+import { RevertOptions, TxOptions } from "../types/contracts.types";
+import { ParseAbiValuesReturnType } from "../types/parseAbiValues.types";
+import { toHexString } from "./toHexString";
+
+/**
+ * Creates a revert data object from revert options
+ */
+export const createRevertData = (
+  revertOptions: RevertOptions
+): RevertOptions => {
+  return {
+    abortAddress: "0x0000000000000000000000000000000000000000", // not used
+    callOnRevert: revertOptions.callOnRevert,
+    onRevertGasLimit: revertOptions.onRevertGasLimit,
+    revertAddress: revertOptions.revertAddress,
+    revertMessage: toHexString(revertOptions.revertMessage),
+  };
+};
+
+/**
+ * Gateway ABI fragments for common functions
+ */
+const GATEWAY_FRAGMENTS = {
```

Comment:
I wonder if we can import ABI from protocol contracts.

Reply:
8173eabc3559bb44e38bccbd7dc93c606b1b29c0